src/encoding.rs

```diff
@@ -77,6 +77,113 @@ where
     }
 }
 
+/// Additional information passed to every decode/merge function.
+#[derive(Clone, Debug)]
+pub struct DecodeContext {
+    #[cfg(feature = "recursion-limit")]
+    /// How many times we can recurse in the current decode stack before we hit
+    /// the recursion limit.
+    ///
+    /// The recursion limit is defined by `RECURSION_LIMIT` and cannot be
+    /// customized. The recursion limit can be ignored by building the Prost
+    /// crate without the `recursion-limit` feature (which is set by default).
```

Comment:
As I understand it, one side effect of modeling this as a on-by-default feature is that if there are two crates in a build graph that use `prost` and one disables it and another does not, it will be not be disabled.  Is that correct?

Alternatively, if it were an opt-in feature to enable turning off the recursion limit, than anyone in the build graph disabling it would disable it for everyone.

I'm not sure which is preferred, but it's an interesting distinction.  Was there a reason to prefer this way as opposed to the other?

Reply:
Interesting, I hadn't thought about that, and instead was thinking only to avoid 'negative variables', but in this case I think it is better to use one.